pkg/apis/serving/v1beta1/explainer_alibi.go

```diff
@@ -66,9 +72,24 @@ func (s *AlibiExplainerSpec) GetContainer(metadata metav1.ObjectMeta, extensions
 		constants.ArgumentModelName, metadata.Name,
 		constants.ArgumentHttpPort, constants.InferenceServiceDefaultHttpPort,
 	}
+
+	argumentPredictorHost := fmt.Sprintf("%s.%s", constants.DefaultPredictorServiceName(metadata.Name), metadata.Namespace)
+	deploymentMode, ok := metadata.Annotations[constants.DeploymentMode]
```

Comment:
We also need to check the global config `deploymentMode` so can use the util function [GetDeploymentMode here](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/utils/utils.go#L68)